Add tax include param to tax REST API

joshuatf opened the pull request:
Fixes #1640 

Adds an `include` param to the tax rest endpoint so we can filter by tax rate IDs and return the correct tax codes.

### Screenshots
<img width="438" alt="screen shot 2019-02-22 at 5 04 55 pm" src="https://user-images.githubusercontent.com/10561050/53231569-360fb600-36c4-11e9-9b50-e05cc546cd49.png">
<img width="515" alt="screen shot 2019-02-22 at 5 04 51 pm" src="https://user-images.githubusercontent.com/10561050/53231570-360fb600-36c4-11e9-943e-40aadbb02bbf.png">


### Detailed test instructions:

1.  Go to the tax report and filter by tax comparison.
2.  Add various tax codes and check that the label is correct even after refreshing.
3.  Test calls to the rest endpoint `wp-json/wc/v4/taxes` with the `include` param and a comma separated list of rate IDs.

joshuatf committed:
Add code to collection params to allow searching

joshuatf committed:
Add include param to tax rest api

Aljullu reviewed (commented):
Before merging this one, we should revert the changes added in #1571 for the taxes `getLabels`, right?
https://github.com/woocommerce/wc-admin/pull/1571/files#diff-2243561b44936466721f610b90a69c51R67

I think it would make sense rebasing this PR with the latest master.

joshuatf committed:
Remove duplicate get_collection_params

joshuatf committed:
Revert to previous taxes getLabel config

joshuatf commented:
>Before merging this one, we should revert the changes added in #1571 for the taxes `getLabels`, right?

@Aljullu Definitely.  Rebased, reverted that, and also updated the includes filters since it looks like we're going to be directly overriding `get_items()` instead of using hooks based on https://github.com/woocommerce/wc-admin/pull/1671.

Aljullu reviewed (approved):
Thanks for rebasing @joshuatf! Code is looking good and it works as expected.
